libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -599,6 +600,36 @@ void AP_OSD::update_current_screen()
             last_switch_ms = 0;
         }
         break;
+    case STICKS_INPUT:
+        auto *runcam = AP::runcam();
+        const uint8_t msg_size = 20;
+        char ret_msg[msg_size] = {0};
+        bool runcam_menu = runcam->pre_arm_check(ret_msg, msg_size);
+        if (!AP_Notify::flags.armed && (!runcam_menu || strcmp(ret_msg, "In OSD menu") != 0)) {
```

Comment:
Please just make is_arming_prevented() public and use that

Reply:
sure

Reply:
done